### PR TITLE
Initialize logging and basic config

### PR DESCRIPTION
**Pull request by smf8**
Hi, I've wanted to suggest this setup for config and logging :D

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "zang"
+version = "0.1.0"
+edition = "2021"
```
Comment (1995parham) [inline Cargo.toml:4]: I think we have 2023 edition 🚀

File: config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+server:
```
Comment (1995parham) [inline config.yaml]: Lets use TOML 😍

File: src/config.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
+use config::{Config, ConfigError};
```
Comment (1995parham) [inline src/config.rs]: Can we move it into its folder? I mean, have `config/mod.rs` instead of `config.rs`.

**Review by 1995parham — CHANGES_REQUESTED** [review]

File: src/config.rs (hunk 3 above)
Comment (smf8) [reply src/config.rs]: Sure.

File: config.yaml (hunk 2 above)
Comment (smf8) [reply config.yaml]: Done ✅

File: Cargo.toml (hunk 1 above)
Comment (smf8) [reply Cargo.toml:4]: the Only newer edition is 2024 which I think is unstable at the moment: 
[Reference](https://doc.rust-lang.org/stable/cargo/reference/unstable.html#edition-2024)

**Pull request merged**